src/Features/Core/Portable/FeaturesResources.resx

```diff
@@ -2808,4 +2808,25 @@ Zero-width positive lookbehind assertions are typically used at the beginning of
   <data name="Convert_to_record" xml:space="preserve">
     <value>Convert to record</value>
   </data>
+  <data name="Introduce_parameter" xml:space="preserve">
+    <value>Introduce parameter</value>
+  </data>
+  <data name="Introduce_parameter_for_0" xml:space="preserve">
+    <value>Introduce parameter for '{0}'</value>
+  </data>
+  <data name="Introduce_parameter_for_all_occurrences_of_0" xml:space="preserve">
+    <value>Introduce parameter for all occurrences of '{0}'</value>
+  </data>
+  <data name="Introduce_overload_with_new_parameter_for_0" xml:space="preserve">
+    <value>Introduce overload with new parameter for '{0}'</value>
+  </data>
+  <data name="Introduce_overload_with_new_parameter_for_all_occurrences_of_0" xml:space="preserve">
+    <value>Introduce overload with new parameter for all occurrences of '{0}'</value>
+  </data>
+  <data name="Extract_method_to_invoke_at_all_callsites_for_0" xml:space="preserve">
+    <value>Extract method to invoke at all callsites for '{0}'</value>
+  </data>
+  <data name="Extract_method_to_invoke_at_all_callsites_for_all_occurrences_of_0" xml:space="preserve">
+    <value>Extract method to invoke at all callsites for all occurrences of '{0}'</value>
```

Comment:
so i've thought about this more, esp. how complex these sentences are.  let's change this to the following:

First, have two *top level* actions:

Introduce parameter for {0}
Introduce parameter for all occurrences of {0}

This is clear and simple to read, and scales well with the size of hte args.  It also pulls up the 'all occurrences' to a single top level item.

Under each of those are 3 items.

and update callsites directly
into extracted method to invoke at callsites
into new overload of {0}

Reply:
this will take us down from 6 items to 3, and will not force the user to wade through the all (or single) cases if they only care about the other.

Reply:
Need to ask about this